### PR TITLE
update user details backend and minor fixes

### PR DESCRIPTION
**Pull request by olukukoyi**
* set up the backend for `settings/profile`
* made changes to `settings/profile` frontend so it now gets current client, makes an API call to get the data for the current user and populate that page with the respective data
* modified signup so that we populate the user's table after creating a new user
* fixed up some minor bugs in the signup form. there were misplaced values which caused some bugs in the signup

### CODE REVIEW

File: topics-project/app/api/update-user-details/route.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+// pass in userID through params
+// .../api/get-user-details/<userID>
+
+export async function POST(request) {
+  const supabase = createClient();
+
+  const data = await request.json();
+
+  console.log(data.uuid);
```
Comment (Abdul-Andha) [inline topics-project/app/api/update-user-details/route.js:12]: leftover console.log

File: topics-project/app/settings/components/ProfileForm.jsx [hunk 2]
```diff
@@ -20,11 +20,34 @@ function ProfileForm() {
...
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          formData,
+          uuid: "3523a764-3925-4f8a-bdac-0436313a2be6",
+        }),
+      }
+    );
+    const data = await response.json();
+    console.log(data);
```
Comment (Abdul-Andha) [inline topics-project/app/settings/components/ProfileForm.jsx:40]: leftover console.log

File: topics-project/app/settings/profile/actions.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { createClient } from "../../../src/utils/supabase/server";
+
+export const fetchUserDetails = async () => {
+  const { data: session, error } = await supabase.auth.getSession();
+
+  if (error) {
+    // Handle error
+  } else if (session) {
+    // Session is active
+    console.log(session);
```
Comment (Abdul-Andha) [inline topics-project/app/settings/profile/actions.js:10]: leftover console.log

File: topics-project/app/settings/profile/actions.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  } else {
+    // No active session
+  }
+  const data = await fetch(
+    "http://localhost:3000/api/get-user-details/3523a764-3925-4f8a-bdac-0436313a2be6"
+  );
+
+  const res = await data.json();
+  console.log(res);
```
Comment (Abdul-Andha) [inline topics-project/app/settings/profile/actions.js:19]: leftover console.log

File: topics-project/app/settings/profile/page.jsx [hunk 5]
```diff
@@ -1,18 +1,59 @@
...
+    const { data, error } = await supabase.auth.getUser();
+
+    if (error) {
+      console.error("Error fetching user:", error.message);
+      router.push("/login");
+    } else if (!data.user) {
+      router.push("/login");
+    } else {
+      setLoading(false);
+      console.log(data.user);
```
Comment (Abdul-Andha) [inline topics-project/app/settings/profile/page.jsx:28]: leftover console.log

File: topics-project/app/settings/profile/page.jsx [hunk 6]
```diff
@@ -1,18 +1,59 @@
...
+      fetchUserDetails(data.user.id);
+    }
+  };
+
+  const fetchUserDetails = async (id) => {
+    const data = await fetch(
+      `http://localhost:3000/api/get-user-details/${id}`
+    );
+    console.log("running!");
```
Comment (Abdul-Andha) [inline topics-project/app/settings/profile/page.jsx:37]: leftover console.log

**Review by Abdul-Andha — COMMENTED** [review]
Looks good, you just have a few console.log statements that should be removed

**Comment by olukukoyi** [comment]
will handle these in a giffy!

**Review by JohanDelao — COMMENTED** [review]
Looks good, some notes:
- `settings` and everything within it must be moved to `\(main)` as its a protected route. 
- In the change profile form, it displays height in inches, however in the form you input it as feet and inches. Would change this in a future PR to maintain consistency.

File: topics-project/app/settings/profile/actions.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import { createClient } from "../../../src/utils/supabase/server";
+
+export const fetchUserDetails = async () => {
```
Comment (JohanDelao) [inline topics-project/app/settings/profile/actions.js:3]: this function seems to be using hardcoded data as well as not handling the cases in the if statements, looks incomplete

**Review by JohanDelao — APPROVED** [review]
LGTM!

File: topics-project/app/settings/profile/actions.js (hunk 7 above)
Comment (olukukoyi) [reply topics-project/app/settings/profile/actions.js]: decided to not make the function in `profile/actions` and to create the function in `profile/page.jsx` instead.usj

i just forgot to delete the file , thx for pointing it out!

**Pull request merged**